Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1991,8 +1991,9 @@ extension BrowserViewController: TabManagerDelegate {
 
     fileprivate func updateTabCountUsingTabManager(_ tabManager: TabManager, animated: Bool = true) {
         let count = tabManager.tabsForCurrentMode.count
-        toolbar?.updateTabCount(count, animated: animated)
-        urlBar.updateTabCount(count, animated: !urlBar.inOverlayMode)
+        // The window check insures that the view is visible and hence can be animated
+        toolbar?.updateTabCount(count, animated: self.view.window != nil ? animated : false)
```

Comment:
This seems like a strange check to be made right here… Is there not a better way we can assure the count will be reflected regardless of animation? (from within `updateTabCount`)

Reply:
Yeah This check wont fix the issue, I realised, I have pushed the fix just now. The animation completion closure was capturing the local count var and thus this issue. Plus made a change to make sure clone tab is not made when its not animated. About the check here, this is also a good to have since its useless to have animation when it will not be visible.

Reply:
Can we remove this now then?

Reply:
Let's just remove the entire animation here. We intentionally removed it in 1.6, and would avoid all of this stuff.